include/net/xdp.h

```diff
@@ -63,17 +63,26 @@ struct xdp_rxq_info {
 	struct xdp_mem_info mem;
 } ____cacheline_aligned; /* perf critical, avoid false-sharing */
 
+#ifdef CONFIG_XDP_LUA
+struct xdplua {
+	struct lua_State	*L;
+	spinlock_t		*lock;
+};
```

Comment:
So this `struct xdplua` is creating another unnecessary level of indirection.
so you already have this:
```
static DEFINE_PER_CPU(spinlock_t, lua_state_lock);
```
And you can easily grab this, as you already do:
```
xdplua->lock = per_cpu_ptr(&lua_state_lock, i);
```
This might be moot if you just remove the lock anyway.

Also, should be possible, and it would be much superior, if instead of having a lua_State pointer as the per cpu attribute, you just had the value directly, that would be one less unnecessary level of indirection.

Reply:
Yes, maybe it would be possible not to use this struct. I'll try this out.